### DISCUSSION TITLE
Monogame game installer and privilege issues

### DISCUSSION THREAD

**Discussion by OtherStuffThatSpam** [MonoGame/MonoGame#8430] [General]
What do you use to make an installer for your games?
I have been using Inno Setup in my past games, and recently one of the users said that after installing my game, it doesn't runs, so he created a new link to my game and ran it as administrator and it worked in Windows 11.

Anyone experienced something similar? I am not sure if the issue is with the installer, the path used (Probably because is a folder in C drive in program files where only the administrator can install and run things) or is something about my game. Anyone has any ideas?

Thanks for any info!

**Comment by Apostolique** [comment]
I've been using [itch.io](https://itch.io/) . They have their [own launcher](https://itch.io/app) which manages updates. For example: <https://apos.itch.io/mitten>.

People can still download the zip file and skip the launcher but the launcher tends to fix issues if they run into any.

**Reply by OtherStuffThatSpam** [reply]
hey thanks! , I will take a look since I am still thinking about how to patches. Though I wasn't looking to integrate with a launcher still this may help. I try not to use Zip files since many people do not want or know how to install zip files

**Comment by KelsamGames** [comment]
I used to use Inno myself before we started publishing with Steam and what not (that obviously handle the installation themselves). There is nothing about Inno specifically that is an issue for you. If your game won't run without admin rights then that's the thing to debug. Run your debug build from the program files (or wherever) folder your customers are installing it to, debug with VS, and see why it is crashing. Almost certainly you'll find your program is trying to read/write some file it isn't allowed to. In which case the fix would be to follow your target platform's guidelines for file read/write. I.e. if you're deploying to Windows systems, you can read files from your own installation's directory but are supposed to put things like save files in specific recommended places like "Documents" or "Users>AppData" folders and what not that your app will have write access to without admin rights. https://learn.microsoft.com/en-us/dotnet/api/system.environment.specialfolder?view=net-8.0

**Reply by OtherStuffThatSpam** [reply]
I tested my game in many folders but it ran properly, so I guess I need to create an account with less privileges and test there. I write sometimes log files in the same game folder when an error happens but if no errors I do not write anything, so the game should still launch, so as you mentioned, I guess the user may be using a lower privilege account and installed into a higher privileged folder, I will test during the weekend. Thanks for the suggestion

**Comment by ThomasFOG** [comment]
Never used an installer and marketplaces don't recommend them. Just zip it, never had anyone complaining about that.

**Comment by stromkos** [comment]
Privilege escalation is never required for a MonoGame game to run under Windows. I rarely use the term "never" unqualified, but I feel it is appropriate here.


`%APPDATA%`  is a good user location for an installer.

User manually unzipping to the Desktop or other user selected location is even better.


If you want to allow an install for all users ask , then file placement in `Program Files` and `C:\users\public\Desktop` require escalation.
